src/NewTokenStandard.ts

```diff
@@ -326,6 +299,17 @@ class FungibleToken extends TokenContract {
       FungibleTokenErrors.noPermissionForSideloadDisabledOperation
     );
 
+    return await this.internalMint(recipient, amount);
+  }
+
+  /**
+   * Internal mint implementation shared by both mint() and mintWithProof().
+   * Contains the core minting logic without proof verification.
+   */
+  private async internalMint(
```

Comment:
Don't know if you considered it, but `async #internalMint` can be used here instead of `private async internalMint` to enforce the private access more completely.

I don't have a strong opinion either way, but I think the `#` syntax would be a good fit here.

Reply:
It makes sense. Do you think that should be the case for `canMint`, `checkPermissionsUpdate` etc. too or only `internalMint`, `internalBurn`, `internalTransfer` and `internalApproveBase`?

Reply:
Both options work here. I would consider consistency and update the other methods if you you do so here, but I don't have a strong preference.